dash-spv-coinjoin/src/coinjoin_client_manager.rs

```diff
@@ -230,33 +232,37 @@ impl CoinJoinClientManager {
                         println!("[RUST] CoinJoin do_automatic_denominating {}", self.str_auto_denom_result);
                         self.last_time_report_too_recent = current_time;
                     }
-                            
-                    self.deq_sessions.push_back(session);
```

Comment:
The biggest problem was actually due to this error. The session was pushed back causing an infinite `while` loop.